src/main/java/it/pagopa/pn/deliverypush/service/impl/IoServiceImpl.java

```diff
@@ -34,24 +39,42 @@ public void sendIOMessage(NotificationInt notification, int recIndex) {
 
         SendMessageRequest sendMessageRequest = getSendMessageRequest(notification, recipientInt);
 
-        ResponseEntity<SendMessageResponse> resp = pnExternalRegistryClient.sendIOMessage(sendMessageRequest);
+        PnAuditLogBuilder auditLogBuilder = new PnAuditLogBuilder();
+        PnAuditLogEvent logEvent = auditLogBuilder.before(PnAuditLogEventType.AUD_AD_SEND_IO, "sendIOMessage")
+                .iun(sendMessageRequest.getIun())
+                .build();
+        logEvent.log();
+        
+        try {
+            ResponseEntity<SendMessageResponse> resp = pnExternalRegistryClient.sendIOMessage(sendMessageRequest);
+
+            if (resp.getStatusCode().is2xxSuccessful()) {
 
-        if (resp.getStatusCode().is2xxSuccessful()) {
-            
-            SendMessageResponse sendIoMessageResponse = resp.getBody();
-            if(sendIoMessageResponse != null){
-                log.info("sendIOMessage OK - iun={} id={} responseId={} with result={}", notification.getIun(), recIndex,
-                        sendIoMessageResponse.getId(), sendIoMessageResponse.getResult() );
-            }else {
-                log.info("sendIOMessage response is not valid response");
-            }
+                SendMessageResponse sendIoMessageResponse = resp.getBody();
+                if(sendIoMessageResponse != null){
+                    if( isErrorStatus( sendIoMessageResponse.getResult() ) ){
+                        logEvent.generateFailure("Error in sendIoMessage, with errorStatus is iun={} id={} ", sendIoMessageResponse.getResult(), notification.getIun(), recIndex).log();
+                    } else {
+                        logEvent.generateSuccess().log();
```

Comment:
nel generate success, logghiamo "cosa" ha fatto ext-registry (così banalmente sappiamo se ha mandato il messaggio di cortesia, di opt-in o se ha "skipppato"